src/qinst/network/NA_N9916A.py

```diff
@@ -204,6 +214,44 @@ def query_data(self, cmd, datatype="REAL,64") -> np.ndarray:
 
         return np.frombuffer(raw_data, dtype=dtype).astype(float)
 
+    def clear_average(self):
+        """Reset averaging."""
+        raise NotImplementedError
+
+    def autoscale(self):
+        """Autoscale selected trace."""
+        raise NotImplementedError
+
+    def read_trace_data(self, yformat=None):
+        """Read trace data from the instrument."""
+        raise NotImplementedError
+
+    def read_freqs(self):
+        """Read trace frequencies from the instrument."""
+        raise NotImplementedError
```

Comment:
```suggestion
    def clear_average(self):
        """Reset averaging."""

    def autoscale(self):
        """Autoscale selected trace."""

    def read_trace_data(self, yformat=None):
        """Read trace data from the instrument."""

    def read_freqs(self):
        """Read trace frequencies from the instrument."""
```

Penso che questo sia legale. Ha senso però solo se controlliamo bene di non istanziare la classe madre.
Eventualmente possiamo anche usare:

```python

from abc import ABC

class NOME(ABC):

    @abstractmethod
    def clear_average(self):
        """Reset averaging."""
```

In questo modo non sarebbe nemmeno necessario controllare che il tipo non sia la classe madre, semplcimente darebbe errore.

Sarei per il secondo metodo